res/res_odbc.c

```diff
@@ -971,17 +972,17 @@ struct odbc_obj *_ast_odbc_request_obj2(const char *name, struct ast_flags flags
 			/* If the connection is dead try to grab another functional one from the
 			 * pool instead of trying to resurrect this one.
 			 */
-			ao2_ref(obj, -1);
-			obj = NULL;
-
 			ast_mutex_lock(&class->lock);
 
 			class->connection_cnt--;
+			ast_cond_signal(&class->cond);
 			ast_debug(2, "ODBC handle %p dead - removing from class '%s', new count is %zd\n",
 				obj, name, class->connection_cnt);
 
 			ast_mutex_unlock(&class->lock);
 
+			ao2_ref(obj, -1);
+			obj = NULL;
```

Comment:
Just moved down to the logging just above will have a non-NULL value for obj.  Of course we can ao2_ref(obj, -1) in original location, and just obj = NULL after logging, but personally I believe these two lines should always stick together.